src/instance/ContainerInstance.ts

```diff
@@ -62,14 +69,18 @@ export default class ContainerInstance implements Searchable, WithAttributes {
     };
   }
 
-  public toXML(parent: Element) {
+  public toXML(parent: Element, options: XMLSerializationOptions = { includeAttributes: false }) {
     const [prefix, href] = this.model.ns;
     const outer = parent.node(this.model.name);
     defineNamespaceOnRoot(parent, prefix, href);
     outer.namespace(prefix);
 
+    if (options.includeAttributes && this.hasAttributes) {
+      this.addAttributes(outer);
+    }
+
     Array.from(this.children.values()).forEach(child => {
```

Comment:
Curious if this is more, less, or on par performance wise with 
```
for (const child of this.children.values())
```
Since it won't have to allocate an array for it to be iterated over